treeherder/push_health/classification.py

```diff
@@ -0,0 +1,111 @@
+                'plugin missing output line for total leaks!',
+                'process() called before end of test suite']
+
+ignore = ['[taskcluster:error]',
+          'Return code:',
+          'ABORT',
+          'Received SIGINT (control-C), so stopped run',
+          'Uncaught exception:',
+          '/builds/worker/workspace/mozharness/mozharness/base/script.py',
+          'self.run_action',
+          'self._possibly_run_method',
+          'Automation Error:',
+          'The mochitest suite:',
+          'TBPL FAILURE',
+          'Assertion failure:',
+          'AsyncShutdown timeout',
+          'Force-terminating active process',
+          'InvalidArgumentException:',
+          'JavascriptException:',
+          'ChunkedEncodingError:',
+          'No suite end',
+          'Logged with data',
+          'No tests run',
+          'HTTP Error 500',
+          'runjunit.py',
+          'runtests.py',
+          'No such file or directory',
+          'not a valid structured log message',
+          'timed out loading test page:',  # raptor
+          'Error copying results',
+          'exception output:',
+          'zombiecheck',
+          'shutil error:',
+          'mozdevice.adb.ADBTimeoutError:',
+          'ValueError: invalid literal for int() with base 10:',
+          'timed out after']
+
+
+def set_classifications(failures, intermittent_history, fixed_by_commit_history):
+    for failure in failures:
+        line_messages = get_line_messages(failure)
+        set_infra(failure, line_messages)
+        set_leak(failure, line_messages)
+        set_intermittent(failure, intermittent_history)
+        set_previous_regression(failure, fixed_by_commit_history)
+
+
+def set_infra(failure, line_messages):
+    if (any([msg in ignore for msg in line_messages]) or
+            failure['testName'] in infra):
+        # TODO: figure out a solution for timeouts/hang on start
+        failure['suggestedClassification'] = 'infra'
+
+
+def set_leak(failure, line_messages):
+    if (any([msg in ignore_leaks for msg in line_messages]) or
+            failure['testName'] in ['leakcheck', 'LeakSanitizer', 'ShutdownLeaks']):
+        # TODO: figure out a solution for leaks, they happen A LOT
+        failure['suggestedClassification'] = 'leak'
+
+
+def set_previous_regression(failure, fixed_by_commit_history):
+    # Not perfect, could have intermittent that is cause of fbc
+    if failure['testName'] in fixed_by_commit_history.keys():
+        failure['suggestedClassification'] = 'previousregression'
+
+
+def set_intermittent(failure, previous_failures):
+    # Not clear if we need these TODO items or not:
+    # TODO: if there is >1 failure for platforms/config, increase pct
+    # TODO: if >1 failures in the same dir or platform, increase pct
+
+    name = failure['testName']
+    platform = failure['platform']
+    config = failure['config']
+    job_name = failure['jobName']
+
```

Comment:
Maybe this could be simplified a bit by iterating through failure to check for the keys?